src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +180,391 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
+                merge_cols=self.extract_data_set_idx_cols(),
+                data_col=scale_col,
+                split_cols=split_cols
+            )
+        # HALP: I want to just assign the output of split_data_on_split_cols
+        # to the frickin scale_col, but it keeps returning a column of nulls
+        # So i'm doing this janky drop and rename
+        scaled_df = (
+            scaled_df.drop(columns=scale_col_dict.keys())
+            .rename(columns={
+                c: c.replace('_scaled', '')
+                for c in [c for c in scaled_df.columns if "_scaled" in c]}
+            )
+        )
+        # STEP 4
+        # second aggregation of the duplicate EIA records.
+        scaled_df_post_agg = self.aggregate_duplicate_eia(
+            connected_to_scale=scaled_df.reset_index(),
+            ppl=ppl
+        )
+        # set the index to be the main EIA plant-part index columns
+        scaled_df_post_agg = (
+            scaled_df_post_agg.set_index(idx_cols + ['record_id_eia'])
+            .reset_index(self.extract_data_set_idx_cols())
+        )
 
-        Note: the following return is the aspirational desire for where this
-        method is going.
+        return scaled_df_post_agg
 
-        Returns:
-            pandas.DataFrame: FERC1 steam data has been either aggregated
-            or disaggregated to match the level of the depreciation records.
-            The data columns properly scaled will be labled as
-            "{data_col}_ferc1_deprish".
-        """
-        self.scale_by_ownership_fraction()
-        same_true = self.assign_ferc1_data_cols_same()
-        same_smol = self.split_ferc1_data_cols()
-        same_beeg = self.agg_ferc_data_cols()
-        logger.info(f"Scaled via same/true: {len(same_true)}")
-        logger.info(f"Scaled via same/smol: {len(same_smol)}")
-        logger.info(f"Scaled via same/beeg: {len(same_beeg)}")
-        scaled_df = pd.concat([same_true, same_smol, same_beeg])
-
-        self.test_same_true_fraction_owned(same_true)
+    def aggregate_duplicate_eia(self, connected_to_scale, ppl):
+        """Aggregate duplicate EIA plant-part records."""
+        dupe_mask = connected_to_scale.duplicated(
+            subset=self.eia_pk, keep=False
+        )
+        # two dfs
+        dupes = connected_to_scale[dupe_mask]
+        non_dupes = connected_to_scale[~dupe_mask]
+        # If there are no duplicate records, then the following aggs will fail
+        # bc there is nothing to merge. If there is a cleaner way to skip this
+        # without a dangly mid-function retrun LMK!
+        if dupes.empty:
+            return non_dupes
         logger.info(
-            f"output is len {len(scaled_df)} while input was "
-            f"{len(self.matches_df)}"
+            f"Aggergating {len(dupes)} duplicate records "
+            f"({len(dupes)/len(connected_to_scale):.1%})")
+
+        # sum and weighted average!
+        de_duped = pudl.helpers.sum_and_weighted_average_agg(
```

Comment:
Is it important that the sum and weighted average aggregation happen at the same time? Are there other contexts in which this happens? It seems like a very specific and kludgy operation that should probably only exist in this module if at all, as opposed to doing the two operations separately.

Reply:
I do these two things in concern enough times across the aggregations that I have done that this function is really helpful. when aggregating numeric data columns i almost always want to do these two things in concert. and it is nice to not have to deal with the merging the outputs of these two operations over and over again.